### DISCUSSION TITLE
Docs on the this.app.vault object?

### DISCUSSION THREAD

**Discussion by mjkanji** [SilentVoid13/Templater#898] [Help] [answered]
I've seen the `this.app.vault` object and its method used to do things like list files in a given vault, etc. Could someone point me to the docs for where this object is coming from and all the methods it has? 

The Templater docs only mention that:

> In script user functions, you can still access global namespace variables like app or moment.

without much explanation beyond this and where more information may be found.

**Comment by AB1908** [comment, marked as answer]
<https://github.com/obsidianmd/obsidian-api/blob/master/obsidian.d.ts>

Obsidian API docs but note that it doesn't correspond 1-to-1. Best try typing app in the console and then begin exploring from there.
